### PR TITLE
fix from_hex bug

### PR DESCRIPTION
**Pull request by zhangjinpeng1987**

### CODE REVIEW

File: src/bin/tikv-ctl.rs [hunk 1]
```diff
@@ -1217,9 +1217,7 @@ fn from_hex(key: &str) -> Vec<u8> {
     const HEX_PREFIX: &str = "0x";
     let mut s = String::from(key);
     if s.starts_with(HEX_PREFIX) {
-        let len = s.len();
-        let new_len = len.saturating_sub(HEX_PREFIX.len());
-        s.truncate(new_len);
+        s.drain(..HEX_PREFIX.len());
```
Comment (siddontang) [inline src/bin/tikv-ctl.rs]: please add a test for it.

**Comment by huachaohuang** [comment]
LGTM

File: src/bin/tikv-ctl.rs (hunk 1 above)
Comment (BusyJay) [reply src/bin/tikv-ctl.rs]: I think you can use `key.trim_left_matches()`.

Comment (AndreMouche) [inline src/bin/tikv-ctl.rs]: Could we use trim_left_matches instead? https://doc.rust-lang.org/std/string/struct.String.html#method.trim_left_matches

File: src/bin/tikv-ctl.rs [hunk 2]
```diff
@@ -1214,12 +1214,9 @@ fn get_module_type(module: &str) -> MODULE {
...
 
 fn from_hex(key: &str) -> Vec<u8> {
-    const HEX_PREFIX: &str = "0x";
     let mut s = String::from(key);
-    if s.starts_with(HEX_PREFIX) {
-        let len = s.len();
-        let new_len = len.saturating_sub(HEX_PREFIX.len());
-        s.truncate(new_len);
+    if s.starts_with("0x") || s.starts_with("0X"){
+        s.drain(..2);
```
Comment (AndreMouche) [inline src/bin/tikv-ctl.rs]: It seems we can use key directly ?
```
 if key.starts_with("0x") || key.starts_with("0X"){
        key = &key[2..];
    }
    key.from_hex().unwrap()
```

**Comment by AndreMouche** [comment]
Why not use `key` instead of `String::from(xxx)`?

**Comment by zhangjinpeng1987** [comment]
@AndreMouche PTAL

File: src/bin/tikv-ctl.rs [hunk 3]
```diff
@@ -1280,3 +1278,16 @@ fn dump_snap_meta_file(path: &str) {
...
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::from_hex;
+
+    #[test]
+    fn test_from_hex() {
+        let result = String::from("t").into_bytes();
```
Comment (huachaohuang) [inline src/bin/tikv-ctl.rs]: Why not just `vec![0x74]`?

**Comment by BusyJay** [comment]
Any updates?

File: src/bin/tikv-ctl.rs [hunk 4]
```diff
@@ -1214,14 +1214,12 @@ fn get_module_type(module: &str) -> MODULE {
 }
 
 fn from_hex(key: &str) -> Vec<u8> {
```
Comment (Hoverbear) [inline src/bin/tikv-ctl.rs]: It seems it is possible for this function to fail. What is the reasoning behind using an `unwrap` and not returning a `Result`?

Comment (BusyJay) [reply src/bin/tikv-ctl.rs]: I guess it's for simplicity.

Comment (breeswish) [reply src/bin/tikv-ctl.rs]: We can make from_hex returning Result<_> and unwraps this result if necessary.

Comment (Hoverbear) [reply src/bin/tikv-ctl.rs]: Maybe it'd better to move this along and create a follow up ticket with this change?

Comment (breeswish) [reply src/bin/tikv-ctl.rs]: @Hoverbear I like this idea. We can also leverage contributors to do this for a warm up!

**Comment by AndreMouche** [comment]
Please fix the conflicts.

**Pull request merged**